src/index.d.ts

```diff
@@ -6,7 +6,14 @@ export function props(elem: any, data?: PropData): void;
 
 export const emit: (elem: any, eventName: string, eventOptions = {}) => boolean;
 
-export class Component extends HTMLElement {}
+export class Component extends HTMLElement {
+  updatedCallback?(prevProps: { [name: string]: any }): boolean;
+  renderCallback?(): void;
```

Comment:
this is wrong definition. `renderCallback` has to return `null` or `JSX.Element` or `vdom`